source/common/router/config_impl.cc

```diff
@@ -1528,7 +1547,9 @@ VirtualHostImpl::VirtualHostImpl(
     hedge_policy_ = virtual_host.hedge_policy();
   }
 
+  shadow_policies_.reserve(virtual_host.request_mirror_policies().size());
   for (const auto& mirror_policy_config : virtual_host.request_mirror_policies()) {
+    validateClusterSpecifier(mirror_policy_config);
```

Comment:
I am thinking the validation is going to check the policy only as well as to keep the function purpose separately. Eventually, this can be achieved through proto static `oneof` check. WDYT?

Reply:
I prefer to call this function inside `ShadowPolicyImpl`, but I can live with this if you and other reviewers also fine with this.

Reply:
Thank you all! Done!